bentoml/service.py

```diff
@@ -299,6 +299,24 @@ def decorator(func):
     return decorator
 
 
+def webui_decorator(static_files):
+    """Define web UI static files required to be bundled with a BentoService
+
+    Args:
+        static_files: path to directory containg index.html and static dir
+
+    >>>  @webui('./ui/')
+    >>>  class MyMLService(BentoService):
+    >>>     pass
+    """
+
+    def decorator(bento_service_cls):
```

Comment:
So I think BentoService needs both

```
@property
def static_files(self):
  [...]
```

as defined above and

```
@static_files.setter
def static_files(self, static_files):
  [...]
```

Obviously, you can do what you're doing just fine--it's Python--but it's nice to define a more solid "contract" in cases like this.

Reply:
> @static_files.setter
> def static_files(self, static_files):
>   [...]
> ```

I am not sure what you mean here, can you please explain a bit more😅

Reply:
https://stackoverflow.com/questions/17330160/how-does-the-property-decorator-work

You have a call somewhere like

```
service._static_files = foo
```

What I am suggesting is two changes: 

- in the class definition of BentoService, add

```
@static_files.setter
def static_files(self, static_files):
  self._static_files = static_files
```

and 

- instead of calling

```
service._static_files = foo
```

call

```
service.static_files = foo
```

Reply:
Ah, ok. I get what you are saying now, but since all other functions are setting in the same way I guess we should be fine. And also it's only sets `_static_files` in one place. 
I'll change it to what you have suggested if needed once maintainers comments come in.

Thanks for the link! I understand decorators much better now!